src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -0,0 +1,263 @@
+Include (%sySecurity, %occErrors, %occReference)
+
+Class %IPM.ResourceProcessor.WebApplication Extends (%IPM.ResourceProcessor.Abstract, %XML.Adaptor) [ PropertyClass = %IPM.ResourceProcessor.PropertyParameters ]
+{
+
+/// Description of resource processor class (shown in UI)
+Parameter DESCRIPTION As STRING = "Facilitates Web Application create/update/delete.";
+
+/// Name of the web application.
+Property Name As %String [ Calculated ];
+
+Method NameGet() As %String
+{
+    Set tName = $Get(..SecurityProperties("Name"), $Get(..SecurityProperties("Url"), ""))
+    If tName = "" {
+        Throw $$$ERROR($$$GeneralError, "Name is required for web applications")
+    }
+    Return tName
+}
+
+/// Keeps track of all application properties
+Property SecurityProperties As %RawString [ MultiDimensional ];
+
+Method CopyAttributes() [ Private ]
+{
+    /*
+     * Overridden to avoid needing a property in this class to correspond to each
+     * attribute as the attribute list is just all non-private/transient properties 
+     * in Security.Applications.
+     */
+    Set key = ""
+    While 1 {
+        Set value = ..ResourceReference.Attributes.GetNext(.key)
+        If (key = "") {
+            Quit
+        }
+        Set eval = ..%Evaluate(value)
+        Set ..SecurityProperties(key) = eval
+    }
+}
+
+Method ReplaceMatchRoles(matchRoles As %String, dbDir As %String) As %String
+{
+    New $NAMESPACE 
+    Set $NAMESPACE = "%SYS"
+    Set templates = $Listbuild("{$dbrole}", "${dbrole}")
```

Comment:
Would add this to the generic %EvaluateSystemExpression() in %IPM.Utils.Module instead of being resource specific.

Reply:
Done!